File: Front/projects/back-office/src/app/Models/event.ts
// event.ts

import { Image } from "./image";

export class Event {
    id!: number;
    type!: EventType; // Assuming EventType is also transformed to TypeScript
    name!: string;
    date!: Date;
    createdAt!: Date;
    createdBy!: number;
    note!: string;
    viewsCounter!: number;
    interestedCounter!: number;
    active!: boolean;
    rating!: number;
    collaboration!: boolean;
    lastModifiedAt!: Date;
    lastModifiedBy!: number;
    depositNotes!: string;
// ------ IMAGE THINGS
      imageUrl?:string
      imageIdCloudinary?:string
imageId?:number;
//-------------------   

interesteds?: Interested[]; // Assuming InterestedBy is another TypeScript interface
    images?: Image[];
    ratings?: Rating[];
    presences?:Presence[];
  }
  
  // Assuming the following interfaces for related entities
  
  export class Interested {
    id!: number;
    userID!: number;
    eventID!: number;
    timestamp!: Date;
  }

  export class Rating {
    id!: number;
    userID!: number;
    eventID!: number;
    rating!: number;
    comment!: string;
    status!:string;
  }
  
// event-type.enum.ts

export enum EventType {
    Enterprise_Day = "Enterprise Day",
    Club_Day = "Club Day",
    Association_Day = "Association Day",
  }
  
  export class Presence {
    id: number;
    eventId: number;
    userId: number;
    date: Date;
  }

File: Back/Events/src/main/java/com/example/Event/Entity/Event.java
package com.example.Event.Entity;


import jakarta.persistence.*;
import lombok.*;

import java.util.Date;
import java.util.Set;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table
@Builder
public class Event {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id ;
//---- the Cloudinary things ----
    private String imageUrl;
    private String imageIdCloudinary;
    private int imageId;
    // -------------------------
    private EventType type;
    @Column( unique = true)
    private String name;
    // event will be at
    private Date date;
    // the event created in database
    private Date created_at;
    // reference USER
    private int created_by;
    private String note;
    // a numberd
    private int viewsCounter;
    // a number
    private int interestedCounter;
    // active is before and during the event date ------ will check it
    private String active;


    //  a rating table to calculate it
    private double rating;
    // cant remove this , it is a checkbox that the event creator MUST select or NOT

    private Date lastModified_at;
    private int lastModified_by;




    /// One to Many    UNIDIRECTIONNEL -----> i dont have accss to the USERs
    @OneToMany(cascade = CascadeType.ALL)
    private Set<Interested> interesteds;

    /// One to Many    UNIDIRECTIONNEL   -- ----  association Galery
    // les personne qui peut voir les stats du event/admin
    @OneToMany(cascade = CascadeType.ALL)
    private Set<Image> images;

    /// One to Many    UNIDIRECTIONNEL
    // la liste des commentaire / ratings
    @OneToMany(cascade = CascadeType.ALL)
    private Set<Rating> ratings;

    /// One to Many    UNIDIRECTIONNEL
    // la liste des commentaire / ratings
    @OneToMany(cascade = CascadeType.ALL)
    private Set<Presence> presences;


}
